src/interp/interp.cc

```diff
@@ -2390,6 +2443,87 @@ RunResult Thread::DoAtomicRmwCmpxchg(Instr instr, Trap::Ptr* out_trap) {
   return RunResult::Ok;
 }
 
+RunResult Thread::DoThrow(Ref exn_ref) {
```

Comment:
I think the purpose of the `Ptr` types is to ensure that your object stays live during interpreter operations, whereas the `Ref` is the representation used for the reftyped value while it lives on the stack, etc. which is trace by the runtime itself.

Yeah as you said it might be more consistent to use a `Ptr` to the exception here. I think in terms of correctness, it should work either way because `DoThrow` immediately creates a `Ptr` to the passed-in `Ref` and ensures that it's live until the end of the function. In addition, an exception is either caught (stored in `exceptions_` and therefore live until popped by control flow out of the catch) or uncaught and doesn't need to be live anymore (this could change if exceptions propagate out to the "host" in a more interesting way in the interpreter).

In any case, I will change this to take a `Ptr` as it's a simple refactoring.

Reply:
I see, thanks. Should we mark `exceptions_` in `Thread::Mark` too? Because apparently `values_` are marked there.

Reply:
Yes, I think you're right. I pushed a small fix for this.